app/Components/App.js

```diff
@@ -1,17 +1,61 @@
 import React from 'react';
 
+import io from 'socket.io-client';
+let socket = io.connect('');
+
 class App extends React.Component {
   constructor(props) {
     super(props);
+    this.joinGame = this.joinGame.bind(this);
+    socket.on('assigngameID', (gameID) => this._assign_game_ID(gameID));
+    socket.on('isgameID', (flag) => this._is_game_ID(flag));
+    this.state = {value: "", warning_on: false};
+    this.handleChange = this.handleChange.bind(this);
+    this.joinGame = this.joinGame.bind(this);
+  }
+
+  joinGame(e) {
+    e.preventDefault();
+    socket.emit('joingame', this.state.value);
+  }
+
+  handleChange(event) {
+    this.setState({value: event.target.value, warning_on: false});
+  }
+
+  _assign_game_ID(gameID) {
+    window.location = '/#/' + gameID;
+  }
+
+  _is_game_ID(flag) {
+    if (flag) {
+      window.location = '/#/' + this.state.value;
+    } else {
+      this.setState({warning_on: true});
+    }
+  }
+
+  createGame(e) {
```

Comment:
I'd suggest naming all event handler methods with something like `onGameCreated`.  The linter rules will check that those methods are in a particular order in the component, which helps find things, and it also indicates that these methods are triggered in response to an event happening, rather than called directly to perform an action.